### PR TITLE
Initial pass at sendfd/recvfd for registry

### PR DESCRIPTION
**Pull request by edwarnicke**
This is quite rough, and could stand some cleaning up,
but works.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -148,8 +155,47 @@ func main() {
 	// ********************************************************************************
 	server := grpc.NewServer(grpc.Creds(grpcfd.TransportCredentials(credentials.NewTLS(tlsconfig.MTLSServerConfig(source, source, tlsconfig.AuthorizeAny())))))
 	endpoint.Register(server)
-	srvErrCh := grpcutils.ListenAndServe(ctx, &config.ListenOn, server)
+	tmpDir, err := ioutil.TempDir("", "forwarder-")
```
Comment (fkautz) [inline main.go:158]: Does this need to be cleaned up after?

**Review by fkautz — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (edwarnicke) [reply main.go:158]: Wouldn't hurt to clean it up... give me a second :)

Comment (edwarnicke) [reply main.go:158]: Fix pushed.

**Review by fkautz — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -148,11 +155,52 @@ func main() {
...
+		config.ConnectTo.String(),
+		grpc.WithTransportCredentials(registryCreds),
+		grpc.WithBlock(),
+	)
+	if err != nil {
+		log.Entry(ctx).Fatalf("failed to connect to registry: %+v", err)
+	}
+
+	registryClient := registrychain.NewNetworkServiceEndpointRegistryClient(
+		registrysendfd.NewNetworkServiceEndpointRegistryClient(),
```
Comment (denis-tingajkin) [inline main.go:182]: Is here a missed `interpose client` chain element?

Comment (edwarnicke) [reply main.go:182]: Why would the interpose client chain element be needed there?

Comment (denis-tingajkin) [reply main.go:182]: Interpose client element makes registration a valid for the interpose server registry

**Pull request merged**